### ISSUE TITLE
Transition to a Hugo Module?

### ISSUE THREAD

**Issue by garrison** [doncabreraphone/victorhugo#4]
Have you considered making victorhugo into a [Hugo Module](https://gohugo.io/hugo-modules/)?  Some examples of this in action are at https://github.com/gohugoio/hugo-mod-jslibs.  I think it might make victorhugo even easier to hook into a web site; it will allow users to more easily keep their version of victorhugo up-to-date; and it will likely mean you can modify victorhugo in ways that could make it more maintainable, rather than have it continue to be one big file.

**Comment by doncabreraphone** [comment]
Too advanced for me. The docunentation seems too cryptic. Maybe in the
future.

**Comment by garrison** [comment]
I agree that the Hugo Modules documentation currently is not very good.  That's why I linked to hugo-mod-jslibs.  For instance, the [wrapping there of turbolinks](https://github.com/gohugoio/hugo-mod-jslibs/tree/master/turbolinks) is pretty simple and straightforward.

**Comment by doncabreraphone** [comment]
I'm giving it a try later today. Thank you for letting me know, I hadn't
seen that bit ypu you mention. That said, I've found the lack of official
support from Hugo to be really demoralizing, to the point where I'm
considering moving the project outside Hugo. Wordpress supported their
plugging devs 100% all the way when they first launched. I don't see the
same from Hugo. It would explain the lack of themes and modules (or
pluggings). Yeah, modules are pretty new but... even then. It is almost
like they can't be bothered with it.

**Comment by doncabreraphone** [comment]
Hot damn James, you are a theoretical physicist! Congrats on you, man!
